core/src/test/java/kafka/test/junit/ClusterTestExtensions.java

```diff
@@ -82,7 +84,9 @@
  * SomeIntegrationTest will be instantiated, lifecycle methods (before/after) will be run, and "someTest" will be invoked.
  *
  */
-public class ClusterTestExtensions implements TestTemplateInvocationContextProvider {
+public class ClusterTestExtensions implements TestTemplateInvocationContextProvider, BeforeEachCallback, AfterEachCallback {
```

Comment:
Maybe this hook should be move to `RaftClusterInvocationContext` and `ZkClusterInvocationContext`. The tests using those clusters are what we care about.

Reply:
It looks like we need to add detection to `ClusterTestExtensions`. I try to make leak thread and use detection in `RaftClusterInvocationContext` and `ZkClusterInvocationContext`, only detection in `ClusterTestExtensions` can work.